### DISCUSSION TITLE
How do I upload and download code from Code Interpreter Dynamic Sessions?

### DISCUSSION THREAD

**Discussion by Kee2u** [microsoft/semantic-kernel#6472] [Q&A] [answered]
https://learn.microsoft.com/en-us/azure/container-apps/sessions-tutorial-semantic-kernel
https://github.com/microsoft/semantic-kernel/tree/main/python/semantic_kernel/core_plugins/sessions_python_tool
I was following these tutorials to work with the new code interpreter functionality in Semantic Kernel. I was able to get a basic chat working but I could not figure out how to upload and download files. Is there any sample code I can reference?

**Comment by moonbox3** [comment, marked as answer]
Hi @Kee2u, I saw you posted a separate issue around using the sessions python plugin, and we fixed the bugs. The `download` method is the only function that is part of the plugin that is not a kernel function. You can still download files from your container, but this needs to be handled manually via code. As for the other kernel functions, they work with chat. In your other issue (#6494) I show a small example of how chatting to upload and list files works. 

Do you need any further help with this? If so, please let us know.

**Reply by Kee2u** [reply]
Hi there @moonbox3,

Yes this is related to the issue so this can be closed. Thank you for the quick resolution!
